crates/store/src/transaction/store_transaction.rs

```diff
@@ -103,10 +103,16 @@ impl StoreTransaction {
     pub fn set_reverted_block_hashes(
         &self,
         reverted_block_smt_root: &H256,
-        block_hashes: Vec<H256>,
+        prev_reverted_block_smt_root: H256,
+        mut block_hashes: Vec<H256>,
     ) -> Result<(), Error> {
         assert!(!block_hashes.is_empty(), "set empty reverted block hashes");
 
+        // Prefix block hashes with prev smt root, order of origin block hashes isn't a matter.
+        block_hashes.push(prev_reverted_block_smt_root);
```

Comment:
If we change the data, we should do it explicit, I suggest to change the return of `get_reverted_block_hashes` to a structure instead of `HashSet<H256>`.

Reply:
Agree.  Add `SMTRevertedBlockHashes` struct.

```rust
pub struct SMTRevertedBlockHashes {
    pub prev_smt_root: H256,
    pub block_hashes: Vec<H256>,
}
```